pkg/api/v1/defaults.go

```diff
@@ -187,6 +187,9 @@ func addDefaultingFuncs(scheme *runtime.Scheme) {
 			if obj.Scheme == "" {
 				obj.Scheme = URISchemeHTTP
 			}
+			if obj.HTTPHeaders == nil {
```

Comment:
I added the test because in the old case the error afterwards, alone, was misleading, because there's no reflect.DeepDiff(). Let me check a bit.